src/python/pants/backend/go/util_rules/build_pkg.py

```diff
@@ -22,26 +22,80 @@
 from pants.util.strutil import path_safe
 
 
-@dataclass(frozen=True)
 class BuildGoPackageRequest(EngineAwareParameter):
-    """Build a package and its dependencies as `__pkg__.a` files."""
-
-    import_path: str
-
-    digest: Digest
-    # Path from the root of the digest to the package to build.
-    subpath: str
-
-    go_file_names: tuple[str, ...]
-    s_file_names: tuple[str, ...]
-
-    # These dependencies themselves often have dependencies, such that we recursively build.
-    direct_dependencies: tuple[BuildGoPackageRequest, ...]
+    def __init__(
+        self,
+        *,
+        import_path: str,
+        digest: Digest,
+        subpath: str,
+        go_file_names: tuple[str, ...],
```

Comment:
Might as well have this be `Iterable[str]` now?

Reply:
I at first did that, then went back to `tuple` because it's not very offensive at the call sites and avoids converting the data type (ack that's a micro-optimization). I'm a little less concerned about an ergonomic API given that this probably won't be consumed by plugin authors, unlike the generic `Process`